src/main/java/org/assertj/core/api/AbstractIterableAssert.java

```diff
@@ -2804,6 +2804,105 @@ private ELEMENT_ASSERT internalElement(int index) {
     return toAssert(elementAtIndex, navigationDescription("element at index " + index));
   }
 
+  /**
+   * Verifies that the {@link Iterable} under test contains a single element and allow to perform assertions that element.
+   * <p>
+   * This is a shorthand for <code>hasSize(1).first()</code>.
+   * <p>
+   * By default available assertions after {@code singleElement()} are {@code Object} assertions, it is possible though to
+   * get more specific assertions if you create {@code IterableAssert} with either:
+   * <ul>
+   * <li>the element assert class, see: {@link Assertions#assertThat(Iterable, Class) assertThat(Iterable, element assert class)}</li>
+   * <li>an assert factory used that knows how to create elements assertion, see: {@link Assertions#assertThat(Iterable, AssertFactory) assertThat(Iterable, element assert factory)}</li>
```

Comment:
Add `singleElement(InstanceOfAssertFactory<?, ASSERT> assertFactory)` as another available option here with a link to the method

Reply:
Thanks for the hint. It has been added.